crates/bevy_ecs/src/system/query.rs

```diff
@@ -298,10 +298,20 @@ where
         };
     }
 
-    /// Runs `f` on each query result in parallel using the given task pool.
+    /// Runs `f` on each query result in parallel using the given [`TaskPool`].
     ///
     /// This can only be called for read-only queries, see [`Self::par_for_each_mut`] for
     /// write-queries.
+    ///
+    ///# Arguments
+    ///
+    ///* `task_pool` - The [`TaskPool`] to use
+    ///* `batch_size` - The items in the query get sorted into batches.
+    /// Internally, this function spawns a group of futures that each take on a `batch_size` sized section of the items (or less if the division is not perfect).
```

Comment:
This detailed description would better fit above the `Arguments` part in my opinion.